addons/mail/static/src/core/common/discuss.scss

```diff
@@ -54,3 +54,8 @@
         }
     }
 }
+
+// Substract the height of the ControlPanel component if it is displayed.
+.o_control_panel + .o-mail-Discuss {
+    max-height: calc(100% - 58px);
+}
```

Comment:
The issue would be resolved with action manager having `d-flex flex-column`.
I think it would be a much better way of doing it.